File: backend/src/vis_platform_backend/contracts/reference_images.py
from datetime import datetime
from typing import Literal

from pydantic import Field

from .common import StrictModel


class ReferenceImageLinks(StrictModel):
    content: str
    thumbnail: str


class ReferenceImage(StrictModel):
    image_id: str
    project_id: str
    name: str
    media_type: Literal["image/png"] = "image/png"
    width: int = Field(gt=0)
    height: int = Field(gt=0)
    byte_size: int = Field(gt=0)
    created_at: datetime
    links: ReferenceImageLinks

File: frontend/src/api/schemas/referenceImages.ts
import { z } from "zod";
import type { components } from "../generated/schema";
import { apiPathSchema } from "./artifacts";

export type ReferenceImage = components["schemas"]["ReferenceImage"];

export const referenceImageSchema: z.ZodType<ReferenceImage> = z
  .object({
    image_id: z.string(),
    project_id: z.string(),
    name: z.string(),
    media_type: z.literal("image/png").default("image/png"),
    width: z.number().int().positive(),
    height: z.number().int().positive(),
    byte_size: z.number().int().positive(),
    created_at: z.string(),
    links: z
      .object({ content: apiPathSchema, thumbnail: apiPathSchema })
      .strict(),
  })
  .strict();
